toolchain/check/context.cpp

```diff
@@ -310,6 +532,13 @@ auto Context::CanonicalizeTypeImpl(
   }()) << "Type was created recursively during canonicalization";
 
   canonical_type_nodes_.InsertNode(type_node_storage_.back().get(), insert_pos);
+
+  // Now we've formed the type, try to complete it and build its value
+  // representation.
+  // TODO: Delay doing this until a complete type is required, and issue a
+  // diagnostic if it fails.
```

Comment:
Maybe record teh other possible TODO around relocating these somewhere so that functions don't have somewhat arbitrary SemIR embedded in them because they happened to be the first one to complete a specific type?

```suggestion
  // TODO: Delay doing this until a complete type is required, and issue a
  // diagnostic if it fails.
  // TODO: Consider emitting this into the file's global node block
  // (or somewhere else that better reflects the definition of the type
  // rather than the coincidental first use).
```